### ISSUE TITLE
User registration page doesn't load

### ISSUE THREAD

**Issue by andyforceno** [twostairs/paperwork#758]
I've successfully installed and configured Paperwork (master branch mostly, develop once) a number of times, but I'm running into an issue with user registration during setup. I'm using the master branch. The page says it's loading the user registration form, but it never loads. In laravel.log I see the following:

```Next exception 'ErrorException' with message 'Undefined variable: admin (View: /var/www/paperwork/frontend/app/views/partials/registration-form.blade.php)' in /var/www/paperwork/frontend/app/storage/views/350367f68925fd45842ade500dc21fc2:55
Stack trace:
#0 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/View/Engines/PhpEngine.php(41): Illuminate\View\Engines\CompilerEngine->handleViewException(Object(ErrorException), 1)
#1 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/View/Engines/CompilerEngine.php(57): Illuminate\View\Engines\PhpEngine->evaluatePath('/var/www/paperw...', Array)
#2 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/View/View.php(140): Illuminate\View\Engines\CompilerEngine->get('/var/www/paperw...', Array)
#3 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/View/View.php(109): Illuminate\View\View->getContents()
#4 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/View/View.php(78): Illuminate\View\View->renderContents()
#5 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Http/Response.php(43): Illuminate\View\View->render()
#6 /var/www/paperwork/frontend/vendor/symfony/http-foundation/Response.php(202): Illuminate\Http\Response->setContent(Object(Illuminate\View\View))
#7 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Router.php(1501): Symfony\Component\HttpFoundation\Response->__construct(Object(Illuminate\View\View))
#8 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Router.php(1036): Illuminate\Routing\Router->prepareResponse(Object(Illuminate\Http\Request), Object(Illuminate\View\View))
#9 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Router.php(1001): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#10 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(781): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#11 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(745): Illuminate\Foundation\Application->dispatch(Object(Illuminate\Http\Request))
#12 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Session/Middleware.php(72): Illuminate\Foundation\Application->handle(Object(Illuminate\Http\Request), 1, true)
#13 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Cookie/Queue.php(47): Illuminate\Session\Middleware->handle(Object(Illuminate\Http\Request), 1, true)
#14 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Cookie/Guard.php(51): Illuminate\Cookie\Queue->handle(Object(Illuminate\Http\Request), 1, true)
#15 /var/www/paperwork/frontend/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Illuminate\Cookie\Guard->handle(Object(Illuminate\Http\Request), 1, true)
#16 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(641): Stack\StackedHttpKernel->handle(Object(Illuminate\Http\Request))
#17 /var/www/paperwork/frontend/public/index.php(65): Illuminate\Foundation\Application->run()
#18 {main}
```

If I click the user registration button, I get the following:
```[2017-02-19 18:22:08] production.ERROR: exception 'ErrorException' with message 'Trying to get property of non-object' in /var/www/paperwork/frontend/app/controllers/UserController.php:27
Stack trace:
#0 /var/www/paperwork/frontend/app/controllers/UserController.php(27): Illuminate\Exception\Handler->handleError(8, 'Trying to get p...', '/var/www/paperw...', 27, Array)
#1 [internal function]: UserController->showRegistrationForm()
#2 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(231): call_user_func_array(Array, Array)
#3 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(93): Illuminate\Routing\Controller->callAction('showRegistratio...', Array)
#4 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(62): Illuminate\Routing\ControllerDispatcher->call(Object(UserController), Object(Illuminate\Routing\Route), 'showRegistratio...')
#5 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Router.php(967): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'UserController', 'showRegistratio...')
#6 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}()
#7 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Route.php(109): call_user_func_array(Object(Closure), Array)
#8 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Router.php(1033): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#9 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Routing/Router.php(1001): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#10 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(781): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#11 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(745): Illuminate\Foundation\Application->dispatch(Object(Illuminate\Http\Request))
#12 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Session/Middleware.php(72): Illuminate\Foundation\Application->handle(Object(Illuminate\Http\Request), 1, true)
#13 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Cookie/Queue.php(47): Illuminate\Session\Middleware->handle(Object(Illuminate\Http\Request), 1, true)
#14 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Cookie/Guard.php(51): Illuminate\Cookie\Queue->handle(Object(Illuminate\Http\Request), 1, true)
#15 /var/www/paperwork/frontend/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Illuminate\Cookie\Guard->handle(Object(Illuminate\Http\Request), 1, true)
#16 /var/www/paperwork/frontend/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(641): Stack\StackedHttpKernel->handle(Object(Illuminate\Http\Request))
#17 /var/www/paperwork/frontend/public/index.php(65): Illuminate\Foundation\Application->run()
#18 {main} 
```

I'm not sure what's going on here, but I suspect it's something I did (or forgot to do), but I can't figure it out. Any suggestions?

**Comment by Liongold** [comment]
A pull request with some fixes to the Registration part of Paperwork is open at #776. Can someone please test if the issues mentioned here are fixed in that pull request, please?

**Comment by andyforceno** [comment]
I'll test it tonight and get back to you.

**Comment by andyforceno** [comment]
Yes, those changes fixed the problem, I was able to register. Thanks!

**Comment by Liongold** [comment]
@andyforceno The pull request has been merged and this feature should be available on both the develop and master branches. Thus, this issue will be closed. Thanks for your feedback. If you encounter any issues, please open a new issue.

**Comment by technoboi** [comment]
I setup paperwork yesterday (6 April) to work on Apache. I still get this problem of the page saying 'loading' and going no further. I also had to setup the preferences manually because that page didn't work either. This is my second install. The first one was via a download. This one was via a git clone. I also found that the 'vendor' directory was missing on the git clone, so I sourced this from the download.
I tried to run an update by running, from 'frontend':-
sudo php artisan paperwork:update
the result was:
_____
{"error":{"type":"ErrorException","message":"Trying to get property of non-object","file":"\/var\/www\/paperwork\/frontend\/app\/config\/database.php","line":41}}
____
Without this registration there appears to be no login screen. I assume this is made dynamically with data loaded from the database. 
I'm about to experiment by populating the database table 'users' to see if that will produce a login page but, this is going to be 'guesswork and hope' to some extent.
